disnake/poll.py

```diff
@@ -0,0 +1,415 @@
+        ----------
+        limit: Optional[:class:`int`]
+            The maximum number of results to return.
+            If ``None``, retrieves every user who voted for this answer.
+            Note, however, that this would make it a slow operation.
+            Defaults to ``100``.
+        after: Optional[:class:`abc.Snowflake`]
+            For pagination, votes are sorted by member.
+
+        Raises
+        ------
+        HTTPException
+            Getting the voters for this answer failed.
+        Forbidden
+            Tried to get the voters for this answer without the required permissions.
+        ValueError
+            You tried to invoke this method on an incomplete object, most likely one that wasn't fetched from the API.
+
+        Yields
+        ------
+        Union[:class:`User`, :class:`Member`]
+            The member (if retrievable) or the user that has voted
+            for this answer. The case where it can be a :class:`Member` is
+            in a guild message context. Sometimes it can be a :class:`User`
+            if the member has left the guild.
+        """
+        if not (self.id is not None and self.poll and self.poll.message):
+            raise ValueError(
+                "This object was manually builded. To use this method you need to get a poll from the discord API!"
+            )
+
+        return PollAnswerIterator(self.poll.message, self.id, limit=limit, after=after)
+
+
+class Poll:
+    """Represents a poll from Discord.
+
+    .. versionadded:: 2.10
+
+    Parameters
+    ----------
+    question: :class:`PollMedia`
+        The question of the poll.
+    answers: List[Union[:class:`str`, :class:`PollAnswer`]]
+        The answers for this poll, up to 10.
+    duration: :class:`datetime.timedelta`
+        The total duration of the poll, up to 32 days. Defaults to 1 day.
+        Note that this gets rounded down to the closest hour.
+    allow_multiselect: :class:`bool`
+        Whether users will be able to pick more than one answer. Defaults to ``False``.
+    layout_type: :class:`PollLayoutType`
+        The layout type of the poll. Defaults to :attr:`PollLayoutType.default`.
+
+    Attributes
+    ----------
+    channel: Optional[Union[:class:`TextChannel`, :class:`VoiceChannel`, :class:`StageChannel`, :class:`Thread`, :class:`DMChannel`, :class:`GroupChannel`, :class:`PartialMessageable`]]
+        The channel that the poll was sent from. This will be None only when the object
+        is builded manually. The library always provide this attribute.
+
+        .. note::
+
+            This attribute is available only if you fetched the Poll object from the API.
+
+    message: Optional[:class:`Message`]
+        The message which contain this poll. This will be None only when the object
+        is builded manually. The library always provide this attribute.
+
+        .. note::
+
+            This attribute is available only if you fetched the Poll object from the API.
+
```

Comment:
```suggestion
    channel: Optional[Union[:class:`TextChannel`, :class:`VoiceChannel`, :class:`StageChannel`, :class:`Thread`, :class:`DMChannel`, :class:`GroupChannel`, :class:`PartialMessageable`]]
        The channel that the poll was sent in. This will be ``None`` only if this object was created manually
        and did not originate from the API.
    message: Optional[:class:`Message`]
        The message which contains this poll. This will be ``None`` only if this object was created manually
        and did not originate from the API.
```
Same deal here, I hope keeping this concise makes it a little easier to understand